webpack/assets/javascripts/react_app/actions/NotificationActions.js

```diff
@@ -1,18 +1,27 @@
 import API from '../API';
 import AppDispatcher from '../dispatcher';
-import {ACTIONS} from '../constants';
+import NotificationsStore from '../stores/NotificationsStore';
+import { ACTIONS, STATUS } from '../constants';
 
 const TIMER = 10000;
 
 export default {
   getNotifications(url) {
-    if (document.visibilityState === 'visible') {
+    if (document.visibilityState !== 'hidden' &&
```

Comment:
@ohadlevy 
I favor changing the 'condition' to 
```js
(document.visibilityState === 'visible'  || document.visibilityState === 'prerender')
```
and so prevent the polling request when page is being unloaded. Does this take care of the turbolinks issue you mentioned?